TarSCM/scm/git.py

```diff
@@ -122,42 +123,17 @@ def detect_version(self, args):
             versionformat = '%ct.%h'
 
         if not parent_tag:
-            cmd = ['git', 'describe', '--tags', '--abbrev=0']
-            try:
-                if args['match_tag']:
-                    cmd.append("--match=%s" % args['match_tag'])
-            except KeyError:
-                pass
-            rc, output = self.helpers.run_cmd(cmd, self.clone_dir)
-
-            if rc == 0:
-                # strip to remove newlines
-                parent_tag = output.strip()
+            parent_tag = self._detect_parent_tag(args)
```

Comment:
Sorry for confusion - Yeah it is some kind of refactoring but it was requested by pylint

`R:115, 4: Too many branches (13/12) (too-many-branches)`

because even it may be confusing, it is consistent in terms of making this modules lint-ready.